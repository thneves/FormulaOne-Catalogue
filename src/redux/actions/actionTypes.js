export const DISPLAY_SEASONS = 'DISPLAY_SEASONS';
export const FILTER_SEASONS = 'FILTER_SEASONS';
