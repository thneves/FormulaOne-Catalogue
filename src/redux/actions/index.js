export const FETCH_SEASONS_REQUEST = 'FETCH_SEASONS_REQUEST';
export const FETCH_SEASONS_SUCCESS = 'FETCH_SEASONS_SUCCESS';
export const FETCH_SEASONS_FAILURE = 'FETCH_SEASONS_FAILURE';
export const FETCH_DRIVERS_REQUEST = 'FETCH_DRIVERS_REQUEST';
export const FETCH_DRIVERS_SUCCESS = 'FETCH_DRIVERS_SUCCESS';
export const FETCH_DRIVERS_FAILURE = 'FETCH_DRIVERS_FAILURE';
export const ADD_FILTER = 'ADD_FILTER';
export const REMOVE_FILTER = 'REMOVE_FILTER';
export const RESET_FILTER = 'RESET_FILTER';
export const DISPLAY_SEASON = 'DISPLAY_SEASON';
export const FETCH_TEAM_REQUEST = 'FETCH_TEAM_REQUEST';
export const FETCH_TEAM_SUCCESS = 'FETCH_TEAM_SUCCESS';
export const FETCH_TEAM_FAILURE = 'FETCH_TEAM_FAILURE';

export const fetchSeasonsRequest = () => ({ type: FETCH_SEASONS_REQUEST });
export const fetchSeasonsSuccess = seasons => ({ type: FETCH_SEASONS_SUCCESS, payload: seasons });
export const fetchSeasonsFailure = error => ({ type: FETCH_SEASONS_FAILURE, payload: error });
export const fetchDriversRequest = () => ({ type: FETCH_DRIVERS_REQUEST });
export const fetchDriversSuccess = drivers => ({ type: FETCH_DRIVERS_SUCCESS, payload: drivers });
export const fetchDriversFailure = error => ({ type: FETCH_DRIVERS_FAILURE, payload: error });
export const addFilter = filter => ({ type: ADD_FILTER, payload: filter });
export const removeFilter = filter => ({ type: REMOVE_FILTER, payload: filter });
export const resetFilter = filter => ({ type: RESET_FILTER, payload: filter });
export const displaySeason = season => ({ type: DISPLAY_SEASON, payload: season });
export const fetchTeamRequest = () => ({ type: FETCH_TEAM_REQUEST });
export const fetchTeamSuccess = team => ({ type: FETCH_DRIVERS_SUCCESS, payload: team });
export const fetchTeamFailure = error => ({ type: FETCH_DRIVERS_SUCCESS, payload: error });
