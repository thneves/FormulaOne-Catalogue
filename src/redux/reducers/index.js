import { combineReducers } from 'redux';
import seasons from './seasons';
import filterSeasons from './filterSeason';

const rootReducer = combineReducers({ seasons, filterSeasons });

export default rootReducer;
