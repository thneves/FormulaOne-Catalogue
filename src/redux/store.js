import { createStore } from 'redux';
import rootReducer from './reducers/index';
/* eslint-disable no-underscore-dangle */
const store = createStore(
  rootReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
);

export default store;
