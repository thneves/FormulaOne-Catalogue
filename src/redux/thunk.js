import store from './store';
import { fetchSeasonsRequest, fetchSeasonsSuccess, fetchSeasonsFailure } from './actions';
import getSeasons from '../requests/api';

const fetchSeasons = () => {
  store.dispatch(fetchSeasonsRequest());
  const requestSeasons = getSeasons();
  requestSeasons.then(seasons => store.disptach(fetchSeasonsSuccess(seasons))).catch(error => {
    store.dispatch(fetchSeasonsFailure(error));
  });
};

export default fetchSeasons;
