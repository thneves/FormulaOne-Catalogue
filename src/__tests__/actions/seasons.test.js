import configureMockStore from 'redux-mock-store';
import { fetchSeasonsRequest, fetchSeasonsSuccess, fetchSeasonsFailure } from '../../redux/actions';

const mockStore = configureMockStore();
const store = mockStore({});

  it('should request the Seasons ranking', () => {
    store.dispatch(fetchSeasonsRequest(
      {
        loading: true,
        seasons: {},
        error: '',
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should fetch seasons with success', () => {
    store.dispatch(
      fetchSeasonsSuccess({
        loading: false,
        seasons: {
          seasons: [2012, 2013]
        },
        error: '',
      })
    );
      expect(store.getActions()).toMatchSnapshot();
  });
  
  
  it('should fetch seasons with failure', () => {
    store.dispatch(
      fetchSeasonsFailure({
        loading: false,
        seasons: {},
        error: 'some error',
      })
    );
    expect(store.getActions()).toMatchSnapshot();
  });

