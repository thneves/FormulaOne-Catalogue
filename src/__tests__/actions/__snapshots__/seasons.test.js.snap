// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fetch seasons with failure 1`] = `
Array [
  Object {
    "type": "FETCH_SEASONS_REQUEST",
  },
  Object {
    "payload": Object {
      "error": "",
      "loading": false,
      "seasons": Object {
        "seasons": Array [
          2012,
          2013,
        ],
      },
    },
    "type": "FETCH_SEASONS_SUCCESS",
  },
  Object {
    "payload": Object {
      "error": "some error",
      "loading": false,
      "seasons": Object {},
    },
    "type": "FETCH_SEASONS_FAILURE",
  },
]
`;

exports[`should fetch seasons with success 1`] = `
Array [
  Object {
    "type": "FETCH_SEASONS_REQUEST",
  },
  Object {
    "payload": Object {
      "error": "",
      "loading": false,
      "seasons": Object {
        "seasons": Array [
          2012,
          2013,
        ],
      },
    },
    "type": "FETCH_SEASONS_SUCCESS",
  },
]
`;

exports[`should request the Seasons ranking 1`] = `
Array [
  Object {
    "type": "FETCH_SEASONS_REQUEST",
  },
]
`;
